docs/design/Pre-UpgradeAnalysis.md

```diff
@@ -0,0 +1,96 @@
+
+## Inter-Project dependencies:
+Upgrade-Assistant has built-in capability to determine inter-project dependencies in a solution and calculate the order in which it is recommended to start upgrade process.
+
+## Surface unsupported API categories:
+If any of the projects use API categories that are not supported in target .NET version like  WCF server APIs, AppDomain creation, WebForms, etc., surface the information to the user.
+**Out Of Scope** : This step will provide a detailed unsupported API list, only the categories that could make the scope of the upgrade process large for the customers.
+
+
+# Analysis for multiple solutions or services:
+
+Customers with a large code base spanning across multiple solutions and services would like to have pre-upgrade analysis for all the solutions, project dependencies between solutions/services (dependency graph) information. In order to support this scenario, upgrade-assistant need to support analysis of multiple solutions / directory containing all the solutions that need to analyzed.
+
+![Control Flow Diagram](../images/design/Pre-UpgradeAnalysis.png)
+
+# Data Visualization and Data Storage:
+For customers needing to store analysis results, scan over scan, across services an option for a separate decentralized solution (possibly a docker windows) that contains data storage option, a local-hosted webservice and data visualization component. Executing the upgrade-assistant analyze command on the host machine would produce the analysis information mentioned in the above sections a format that then the user can ingest into the data storage.
+
+Alternately customers can also use their own storage solution and ingest analysis data from Upgrade-Assistant into their data storage.
+
+Customers can choose to integrate upgrade-assistant analysis to their CI/CD system (AzDO/GH Actions) that then can do the analysis for a solution/service on a specified cadence.
+
+Detailed instructions on how to set up the decentralized solution will be provided via dotnet blog and readme in the repo.
+
+# Implementation Phases:
+
+## Phase I:
+- Add analyze command capabilities to upgrade-assistant to support a single solution / entry-point. This is the level of entry-point that is already supported for the upgrade path. 
+- Provide an output format (XML/JSON/HTML/XLSX) for the analyzed results.
+
+**Cost** : L
+
+## Phase II: 
+- Producing AzDO pipeline tasks/GH actions/etc that could run in the context of a build and produce analysis artifacts that then can be consumed by a data store.
+
+**Cost** : M
+
+## Phase III:
+Add loose assembly analysis capability to upgrade-assistant.
+- Chem integration.
+
+**Cost** : L (This is L coz this requires quite a bit of experimental prototyping to know the best way to integrate Chem. This effort can happen in parallel to Phase I and Phase II)
+
+## Phase IV:
+- Analyze command supports multiple solutions / directory of solutions.
```

Comment:
This could easily be combined into a CI/CD system for Phase II without changing the tool itself